### ISSUE TITLE
Enhanced docstring for callFromThread

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#11208]
|<img alt="moonfallen's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| moonfallen reported|
|-|-|
|Trac ID|trac#1726|
|Type|enhancement|
|Created|2006-05-18 23:42:23Z|

"""
[ ... ] and I'm confused every time I see those functions. It's probably too late to change the name, but might I suggest changing the first sentence of callFromThread's docstring from:

    Call a function from within another (i.e. non-reactor) thread.

to:

    Cause a function to be executed asynchronously on the reactor thread. 
"""

It's pretty silly that I have to create a branch to do this, but okay.

<details><summary>Searchable metadata</summary>

```
trac-id__1726 1726
type__enhancement enhancement
reporter__moonfallen moonfallen
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1147995743000000 1147995743000000
changetime__1154196187000000 1154196187000000
version__ 
owner__ 
cc__moonfallen cc__exarkun
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="moonfallen's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|moonfallen commented|
|-|-|

Reassigning randomly, per Glyph.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun commented|
|-|-|

For future reference, you could just attach a patch to the ticket if you want.

**Comment by twisted-trac** [comment]
|<img alt="moonfallen's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|moonfallen commented|
|-|-|

glyph doesn't want it, let's try dash.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph set owner to dash|
|-|-|

Reassigning rather than ccing

**Comment by twisted-trac** [comment]
|<img alt="moonfallen's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|moonfallen commented|
|-|-|

BTW branch is docstring-1726.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph set owner to moonfallen|
|-|-|

I definitely see the problem with the existing text, but "Cause a function to be executed asynchronously on the reactor thread" could also mean "when running on the reactor thread, cause a function to be executed asynchronously (e.g. in a thread).

The .NET platform uses "asynchronously" to mean run in a thread with a callback you can request in your own thread.

Let's come up with another phrasing...

**Comment by twisted-trac** [comment]
|<img alt="moonfallen's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|moonfallen commented|
|-|-|

Glyph suggests this instead of the first two paragraphs: 

        """From code running within a non-reactor thread (e.g. one created by
        callInThread), call this method to cause the given function to be
        invoked at some point in the future by the reactor on the main I/O
        thread."""

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun commented|
|-|-|

What's up with the win32/maintainer-nodes.txt changes in this branch?

The above docstring is inaccurate, so it shouldn't be used.  Dropping "asynchronously" from the docstring currently in svn and replacing "on" with "by" seems fine to me.

**Comment by twisted-trac** [comment]
|<img alt="moonfallen's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|moonfallen set status to `closed`|
|-|-|

(In [17736]) Resolve bug [#11208](https://github.com/twisted/twisted/issues/11208), edit docstring for callFromThread

Fixes [#11208](https://github.com/twisted/twisted/issues/11208)

Author: moonfallen

Reviewer: exarkun

Candy: delicious

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
